src/test/java/io/appium/java_client/localserver/ServerBuilderTest.java

```diff
@@ -1,434 +1,311 @@
+import static io.appium.java_client.service.local.AppiumServiceBuilder.APPIUM_PATH;
+import static io.appium.java_client.service.local.flags.GeneralServerFlag.CALLBACK_ADDRESS;
+import static io.appium.java_client.service.local.flags.GeneralServerFlag.PRE_LAUNCH;
+import static io.appium.java_client.service.local.flags.GeneralServerFlag.SESSION_OVERRIDE;
+import static java.lang.System.setProperty;
+import static java.util.Arrays.asList;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import io.appium.java_client.service.local.AppiumDriverLocalService;
+import io.appium.java_client.service.local.AppiumServiceBuilder;
+import org.apache.commons.validator.routines.InetAddressValidator;
+import org.junit.After;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.openqa.selenium.Platform;
+import org.openqa.selenium.remote.DesiredCapabilities;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.OutputStream;
+import java.net.InetAddress;
+import java.net.NetworkInterface;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.Properties;
+
+public class ServerBuilderTest {
+
+    private static Properties properties;
+    private static String testIP;
+    private AppiumDriverLocalService service;
+    private File file;
+    private OutputStream stream;
+
+
+    /**
+     * initialization.
+     */
+    @BeforeClass public static void beforeClass() throws Exception {
+        File file =
+            new File("src/test/java/io/appium/java_client/localserver/custom_node_path.properties");
+        FileInputStream fileInput = new FileInputStream(file);
+        properties = new Properties();
+        properties.load(fileInput);
+        fileInput.close();
+
+        for (Enumeration<NetworkInterface> en = NetworkInterface.getNetworkInterfaces(); en
+            .hasMoreElements(); ) {
+            NetworkInterface intf = en.nextElement();
+            for (Enumeration<InetAddress> enumIpAddr = intf.getInetAddresses(); enumIpAddr
+                .hasMoreElements(); ) {
+                InetAddress inetAddress = enumIpAddr.nextElement();
+                if (!inetAddress.isLoopbackAddress()) {
+                    InetAddressValidator validator = InetAddressValidator.getInstance();
+                    testIP = inetAddress.getHostAddress().toString();
+                    if (validator.isValid(testIP)) {
+                        break;
+                    }
+                    testIP = null;
+                }
+            }
+            if (testIP != null) {
+                break;
+            }
+        }
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        if (service != null) {
+            service.stop();
+        }
+        if (stream != null) {
+            stream.close();

```

Comment:
No need to se values to null in @After method. JUnit will make a new instance of ServerBuilderTest for each @Test which will automatically reset all member fields.